src/main/java/net/mcreator/ui/modgui/FluidGUI.java

```diff
@@ -292,6 +292,13 @@ public FluidGUI(MCreator mcreator, ModElement modElement, boolean editingMode) {
 		return fluid;
 	}
 
+	@Override protected void afterGeneratableElementStored() {
+		super.afterGeneratableElementStored();
+		modElement.clearMetadata();
+		modElement.putMetadata("gb", generateBucket.isSelected());

```

Comment:
You will need to add a fixer so existing fluids get "gb" set. Check DimensionPortalSelectedFixer how I did this for portals.